sdks/python/src/opik/api_objects/experiment/rest_operations.py

```diff
@@ -12,25 +13,25 @@ def get_experiment_data_by_name(
     #  deprecated Opik.get_experiment_by_name() will be removed.
     #  This function should not be used anywhere else except for deprecated logic as it is confusing and misleading
 
-    while True:
-        experiment_page_public = rest_client.experiments.find_experiments(name=name)
-        if len(experiment_page_public.content) == 0:
-            raise exceptions.ExperimentNotFound(
-                f"Experiment with the name {name} not found."
-            )
+    experiments = get_experiments_data_by_name(rest_client, name)
+    for experiment in experiments:
+        if experiment.name == name:
+            return experiment
 
-        for experiment in experiment_page_public.content:
-            if experiment.name == name:
-                return experiment
+    raise exceptions.ExperimentNotFound(f"No experiment found with the name '{name}'.")
 
 
 def get_experiments_data_by_name(
     rest_client: OpikApi, name: str
 ) -> List[experiment_public.ExperimentPublic]:
-    experiment_page_public = rest_client.experiments.find_experiments(name=name)
-    if len(experiment_page_public.content) == 0:
+    experiments_stream = rest_client.experiments.stream_experiments(name=name)
```

Comment:
we have a 2k items limit per stream, so we should call this function until we are sure there are no more experiments (take a look at other places where `read_and_parse_stream` is called)